mimic/test/behavior_tests.py

```diff
@@ -0,0 +1,307 @@
+            behavior_ids = []
+            for i, n_and_p in enumerate(names_and_params):
+                name, params = n_and_p
+                behavior_ids.append(register_behavior(
+                    self,
+                    self.bhelper.root,
+                    self.bhelper.behavior_api_endpoint,
+                    name,
+                    params,
+                    self.bhelper.criteria))
+
+            self.bhelper.validate_injected_behavior(
+                names_and_params[0], *self.bhelper.trigger_event())
+
+            self.delete_behavior(behavior_ids[0])
+
+            if len(behavior_ids) > 1:
+                self.bhelper.validate_injected_behavior(
+                    names_and_params[1], *self.bhelper.trigger_event())
+
+                self.delete_behavior(behavior_ids[1])
+
+            self.bhelper.validate_default_behavior(
+                *self.bhelper.trigger_event())
+
+    Tester.__name__ = "TestsFor{0}".format(behavior_helper_factory.name)
+    Tester.__module__ = behavior_helper_factory.module
+    return Tester
+
+
+def behavior_tests_helper_class(klass):
+    """
+    Generate a test suite containing test that validate that:
+
+    - deleting a behavior will revert to the underlying
+      behavior.  If 2 behaviors are provided, will also assert that the
+      first behavior registered will be the behavior used, and that deleting
+      it means the second mock takes over.  Deleting that will revert to
+      default behavior.  If only 1 behavior is provided, it will just test
+      that deleting it reverts to default behavior.
+
+    - deleting an invalid behavior for will result in a 404.
+
+    - providing invalid JSON will result in a 400 when creating the behavior.
+
+    A basic version of ``klass`` should have all the methods and attributes
+    required by :class:`IBehaviorAPITestHelper`, and an `__init__` function
+    that takes a :class:`twisted.trial.unittest.SynchronousTestCase`.
+
+    This decorator is syntactic sugar for
+    #. declaring that ``klass`` implements :class:`IBehaviorAPITestHelper` if
+       it hasn't already been declared, and verifying that it does
+    #. assigning a ``name`` and ``module`` attribute to ``klass`` if they
+       aren't assigned already declaring (and validating that)
+    #. setting ``from_test_case`` to be a method that calls the ``klass``
+       initializer with a test case if ``from_test_case`` is not already
+       provided
+    #. declaring that ``klass`` also implements
+       :class:`IBehaviorAPITestHelperFactory`
+    #. calling :func:`make_behavior_tests` on ``klass``
+
+    This decorator will also validate that ``klass`` correctly implements
+    :class:`IBehaviorAPITestHelper`.  It creates an instance of ``klass`` to
+    do so.
+
+    :param klass: a class that implements :class:`IBehaviorAPITestHelper`,
+        although it does not have to declare that it does (this decorator will
```

Comment:
Question: should this require that `klass` already be declared a `IBehaviorAPITestHelper`?